### ISSUE TITLE
offline and adaptable legend 

### ISSUE THREAD

**Issue by letypequividelespoubelles** [osmandapp/Osmand#7441]
As map is offline, it would be nice to have offline legend too, and easier to see it (maybe with quick action? )
Also, it would be wonderful to have adaptable legend, I mean legend of just what is now shown on the map. Possible to do it? 

Before entering an issue, please read the information below.

```GitHub is a main development tool for developers. There are hundreds requests a month and there are only few developers. So by opening an issue, please know that your issue will be sent out to all developers and acknowledge that it could be closed without explanation or with a brief message. Comments on the closed issues are also sent to all developers, so you will definitely will be heard. However, there is no guarantee that a developer will pick up the issue to work on it.```

In case you are not sure to open issue or not, please send your request to support support@osmand.net. For **general questions** about OsmAnd, please use the [Google group](https://groups.google.com/forum/#!forum/osmand).

Please be sure to have read our [FAQ](http://osmand.net/help-online) before creating an issue here. (Also, tell us in Google Groups when the FAQ is wrong or outdated.)

The best way to get help about an OsmAnd issue is to create a valid and detailed issue content.
Please give us the following information so that we can try to **reproduce** your issue:

What version of OsmAnd are you using, on what device and Android/iOS version?

Do you use the offline maps offered within the OsmAnd app for download, or online (tile / raster) maps?

If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs, and its edition date.

__*In case of wrong routing:*__

Tell us whether you have used OsmAnd's in-app offline routing, or any online routing provider like YOURS, OpenRouteService or OSRM.

What routing profile is chosen in OsmAnd app? (car, bike, pedestrian, fastest or shortest)

Please specify as exactly as possible the start and end point of your route by telling us city name and street name so that we can find it via in-app address search easily. Also, a permalink from openstreetmap.org can be helpful.

Tell us your expected routing, and how OsmAnd routes, or add screenshots here.

**Comment by pietervdvn** [comment]
Under the menu (swipe from the left), select Help.

There is a title 'Map Legend' there - although not dynamically generated.

**Comment by letypequividelespoubelles** [comment]
Yes, I know but :
- you need internet connection, while offline you can't see the image
- it's not adaptable to what is on the screen, which can be useful

**Comment by letypequividelespoubelles** [comment]
And also for the moment, legend is shown only for osmand default map style, ski and sea map. Nothing for lightRS, topo and so on.

**Comment by OmlineEditor** [comment]
you need the help to be cached, then you can look at it offline, it is advisable to cache the entire help from the menu.

**Comment by mariush444** [comment]
legend plugin works offline, nothing has to be cashed

**Comment by OmlineEditor** [comment]
> legend plugin works offline, nothing has to be cashed

I don't need plugins. I need to go to: **_main menu => help_**
in this menu I want to watch pictures. in particular, I need to look at the navigation instructions, FAQ and legend. Now these images are downloaded every time, this is an increased load on the phone and your server. I ask you to download images for the help menu 1 time and leave them in the phone cache. OK?

**Comment by mariush444** [comment]
I'm just a regular user so I think I can't help you in other way.
But I don't like idea that everything will be off line on my phone like FAQ or help.

**Comment by fabeit** [comment]
> > legend plugin works offline, nothing has to be cashed
> 
> I don't need plugins. I need to go to: **_main menu => help_** in this menu I want to watch pictures. in particular, I need to look at the navigation instructions, FAQ and legend. Now these images are downloaded every time, this is an increased load on the phone and your server. I ask you to download images for the help menu 1 time and leave them in the phone cache. OK?

That's exactly the point which I would like the developers to provide the reason why these things are not cached off-line. We have hundreds Gb of useless files stored in our phones and we cannot add a few kilobytes for a legend?

**Comment by vshcherb** [comment]
That's true we plan to update documentation and all images should be stored of course but we are in the middle of changing documentation engine.

**Comment by mnalis** [comment]
Has documentatin engine been changed now? Can  help / legend be made cacheable now, so they work offline ?

**Comment by DmitryAlexei** [comment]
@mnalis In latest nightly builds documentation can be cashed and used offline (Menu > Help)
